API/common/service_impl/chatgpt.py

```diff
@@ -33,23 +33,30 @@ def gpt_generate(user_msg: str, history_msg: List[Dict[str, str]]) -> Dict[str,
         response = openai.ChatCompletion.create(**chatgpt_service_params)
     except (TryAgain, Timeout) as e:
         raise GPTTimeOutException() from e
+    except RateLimitError as e:
+        raise CopilotException('The OpenAI API rate limit is exceeded.') from e
     except OpenAIError as e:
         raise CopilotException('There is some error from the OpenAI.') from e
     content = response["choices"][0]["message"]["content"]
     if "out of my scope" in content and content.startswith("Sorry"):
         return None
     try:
-        content = json.loads(content.replace("\"", "\\\"").replace("'", '"'))
+        # If only single quotes exist, replace them with double quotes.
+        if "'" in content and not '"' in content:
+            content = json.loads(content.replace("'", '"'))
+        # In other cases, convert directly to json
+        else:
+            content = json.loads(content)
```

Comment:
Can the parsing logic handle cases where the argument value is a JSON object?
For example, 
```json
{
    "command": "az stream-analytics input create",
    "arguments": [
        "--name",
        "--resource-group",
        "--job-name",
        "--type",
        "--serialization",
        "--datasource"
    ],
    "reason": "Create an input for the Stream Analytics job",
    "example": "az stream-analytics input create --name MyInput --resource-group MyResourceGroup --job-name MyJob --type Stream --serialization '{\"type\": \"Json\",\"properties\": {\"encoding\": \"UTF8\"}}' --datasource '{\"type\": \"Microsoft.Storage/Blob\",\"properties\": {\"storageAccounts\": [{\"accountName\": \"MyAccount\",\"accountKey\": \"MyKey\"}],\"container\": \"MyContainer\",\"pathPattern\": \"telemetry/{date}/{time}\"}}'",
    "result": [],
    "example_result": []
},
```